src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/BytesExtensions.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System.Text;
+using System;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions
+{
+    internal static class BytesExtensions
+    {
+        /// <summary>
+        /// Converts a byte array to Base64URL string with optional padding ('=') characters removed.
+        /// Base64 description: https://datatracker.ietf.org/doc/html/rfc4648.html#section-4
+        /// </summary>
+        public static string ToBase64Url(this byte[] bytes)
+        {
+            string bytesBase64 = Convert.ToBase64String(bytes);
+
+            int indexOfEquals = bytesBase64.IndexOf("=");
+
+            // Remove all instances of "=" at the end of the string that were added as padding
+            int stringBuilderCapacity = indexOfEquals != -1 ? indexOfEquals : bytesBase64.Length;
```

Comment:
This code trims everything after the first occurrence of the `=`. It doesn't match your comment. You assume the `=` is at the end of the string. I'm wondering why we don't simply use `TrimEnd`.

Reply:
Base64 doesn't use equals as any value other than padding at the end, so it should work effectively the same. I had it as `TrimEnd` initially but I got lost with this other implementation that was suggested and I didn't reconsider it. I think it makes more sense to be `TrimEnd` to save iteration and be more clear. Jumped the gun on the merge, so I can make another PR.

Reply:
I don't see why TrimEnd would be used, it's unnecessary allocation. Otherwise it would just be `Convert.ToBase64String(bytes).Replace('+', '-').Replace('/', '_').TrimEnd('=')`. But that creates copies a bit carelessly.

Reply:
Implementation is fine. The comment can be updated to clarify that we choose to skip the padding (if any) based on the spec.